cups/file.c

```diff
@@ -335,6 +335,131 @@ _cupsFileCheckFilter(
 #endif /* !_WIN32 */
 
 
+/*
+ * 'cupsFileGetConfAndComments()' - Get line and comments from a configuration file.
+ */
+
+char *					/* O  - Line read or @code NULL@ on end of file or error */
+_cupsFileGetConfAndComments(cups_file_t *fp,	/* I  - CUPS file */
+                char        *buf,	/* O  - String buffer */
+		size_t      buflen,	/* I  - Size of string buffer */
+                char        **value,	/* O  - Pointer to value */
+		int         *linenum)	/* IO - Current line number */
+{
+  char	*ptr;				/* Pointer into line */
+
+
+ /*
+  * Range check input...
+  */
+
+  DEBUG_printf(("2cupsFileGetConfAndComments(fp=%p, buf=%p, buflen=" CUPS_LLFMT
+                ", value=%p, linenum=%p)", (void *)fp, (void *)buf, CUPS_LLCAST buflen, (void *)value, (void *)linenum));
+
+  if (!fp || (fp->mode != 'r' && fp->mode != 's') ||
+      !buf || buflen < 2 || !value)
+  {
+    if (value)
+      *value = NULL;
+
+    return (NULL);
+  }
+
+ /*
+  * Read the next line...
+  */
+
+  *value = NULL;
+
+  while (cupsFileGets(fp, buf, buflen))
+  {
+    (*linenum) ++;
+
+    /*
+     * Strip the comment from inline comment...
+     */
+     if ((ptr = strchr(buf, '#')) != NULL)
+     {
+       int index = (int) (ptr - buf);
+       while (index > 0 && _cups_isspace(buf[index]))
+         buf[index--] = '\0';
+     }
+
+     /*
+      * Strip leading whitespace...
+      */
+
+      for (ptr = buf; _cups_isspace(*ptr); ptr ++);
+
+      if (ptr > buf)
+        _cups_strcpy(buf, ptr);
+
+      /*
+       * See if there is anything left...
+       */
+
+       if (buf[0])
+       {
+         // return comment if any
+         if (buf[0] == '#')
+           return ptr;
+
+        /*
+         * Otherwise grab any value and return...
+         */
+
+         for (ptr = buf; *ptr; ptr ++)
+           if (_cups_isspace(*ptr))
+             break;
+
+         if (*ptr)
+         {
+          /*
+           * Have a value, skip any other spaces...
+           */
+
+           while (_cups_isspace(*ptr))
+             *ptr++ = '\0';
+
+           if (*ptr)
+             *value = ptr;
+
+          /*
+           * Strip trailing whitespace and > for lines that begin with <...
+           */
+
+           ptr += strlen(ptr) - 1;
+
+           if (buf[0] == '<' && *ptr == '>')
+             *ptr-- = '\0';
+           else if (buf[0] == '<' && *ptr != '>')
+           {
+            /*
+             * Syntax error...
+             */
+
+             *value = NULL;
+             return (buf);
+           }
+
+           while (ptr > *value && _cups_isspace(*ptr))
+             *ptr-- = '\0';
+         }
+
+         /*
+          * Return the line...
+          */
+
+          return (buf);
+       }
+       else
```

Comment:
The snippet I've provided in the previous review was pseudocode - not actual code which you can put into the code and have it done, but a code which is supposed to express an idea and has to be adjusted for actual source code.
Additionally I mentioned the if-else structure you introduced in the previous commit is redundant. Think about which part could be removed.